mcp/runtime/src/main/java/io/quarkiverse/langchain4j/mcp/runtime/http/QuarkusHttpMcpTransport.java

```diff
@@ -0,0 +1,189 @@
+package io.quarkiverse.langchain4j.mcp.runtime.http;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+import java.io.IOException;
+import java.net.URI;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.client.api.LoggingScope;
+import org.jboss.resteasy.reactive.server.jackson.JacksonBasicMessageBodyReader;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import dev.langchain4j.mcp.client.protocol.CancellationNotification;
+import dev.langchain4j.mcp.client.protocol.McpCallToolRequest;
+import dev.langchain4j.mcp.client.protocol.McpClientMessage;
+import dev.langchain4j.mcp.client.protocol.McpInitializeRequest;
+import dev.langchain4j.mcp.client.protocol.McpListToolsRequest;
+import dev.langchain4j.mcp.client.transport.McpOperationHandler;
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import io.quarkus.rest.client.reactive.QuarkusRestClientBuilder;
+
+public class QuarkusHttpMcpTransport implements McpTransport {
+
+    private static final Logger log = Logger.getLogger(QuarkusHttpMcpTransport.class);
+    private final String sseUrl;
+    private final McpSseEndpoint sseEndpoint;
+    private final Duration timeout;
+    private final boolean logResponses;
+    private final boolean logRequests;
+    private SseSubscriber mcpSseEventListener;
+
+    // this is obtained from the server after initializing the SSE channel
+    private volatile String postUrl;
+    private volatile McpPostEndpoint postEndpoint;
+    private volatile McpOperationHandler operationHandler;
+
+    public QuarkusHttpMcpTransport(QuarkusHttpMcpTransport.Builder builder) {
+        sseUrl = ensureNotNull(builder.sseUrl, "Missing SSE endpoint URL");
+        timeout = getOrDefault(builder.timeout, Duration.ofSeconds(60));
+
+        this.logRequests = builder.logRequests;
+        this.logResponses = builder.logResponses;
+
+        QuarkusRestClientBuilder clientBuilder = QuarkusRestClientBuilder.newBuilder()
+                .baseUri(URI.create(builder.sseUrl))
+                .connectTimeout(timeout.toSeconds(), TimeUnit.SECONDS)
+                .readTimeout(timeout.toSeconds(), TimeUnit.SECONDS)
+                .loggingScope(LoggingScope.ALL)
+                .register(new JacksonBasicMessageBodyReader(new ObjectMapper()));
```

Comment:
Do we need a brand new `ObjectMapper`? Asking because these are very heavy objects. Can't we use `ObjectMapperHolder.MAPPER`?